src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -884,6 +884,19 @@ public virtual int GetByteCount(String s)
         [Pure]
         public abstract int GetByteCount(char[] chars, int index, int count);
 
+        // Returns the number of bytes required to encode a string range.
+        //
+        [Pure]
+        public virtual int GetByteCount(string s, int index, int count)
+        {
+            if (s == null)
```

Comment:
@AlexRadch I assume you are going to change the implementation to use the unsafe code instead of the allocation as @jkotas mentioned. right?